### PR TITLE
improvement/general-housekeeping

### PR DESCRIPTION
**Pull request by dabeeeenster**
- Update versions for some github actions that were deprecated
- Linting / precommit for yaml
- Added prettier
- Fixed up some files

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Ignore a bunch of files:
```
Comment (matthewelwell) [inline .prettierignore:1]: This comment doesn't add any additional information that can't be inferred from the name of the file. Either improve the comment to explain _why_ we're ignoring, or just remove. I don't mind which.

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
-            - name: Run Tests
-              run: poetry run pytest
+  test:
+    runs-on: ubuntu-latest
+    name: Pytest and Black formatting
+
+    strategy:
+      max-parallel: 4
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10']
```
Comment (matthewelwell) [inline .github/workflows/pytest.yml:14]: We should drop 3.7 from here and add 3.11 but I think @kim has done this in another PR already?

File: .prettierignore (hunk 1 above)
Comment (dabeeeenster) [reply .prettierignore]: OK yep - removed the ignore file and linted some html and json

File: .github/workflows/pytest.yml (hunk 2 above)
Comment (dabeeeenster) [reply .github/workflows/pytest.yml]: Couldnt find that PR so updated this file

**Review by matthewelwell — APPROVED** [review]
